server/src/main/java/org/elasticsearch/transport/RemoteClusters.java

```diff
@@ -89,7 +89,12 @@ public synchronized Client getClient(String subscriptionName) {
             remoteClusters.put(name, remoteCluster);
             return remoteCluster.connectAndGetClient();
         } else {
-            return CompletableFuture.completedFuture(remoteCluster.client());
+            connectedClient = remoteCluster.client();
```

Comment:
What do you think of always calling `connectAndGetClient` here? It's internally re-using a client if it exists:

https://github.com/crate/crate/blob/afcaa6f638a8efca5064099b8178724e65ecf24f/server/src/main/java/org/elasticsearch/transport/RemoteCluster.java#L119-L124

And maybe add a nullable annotation to the client property?

https://github.com/crate/crate/blob/afcaa6f638a8efca5064099b8178724e65ecf24f/server/src/main/java/org/elasticsearch/transport/RemoteCluster.java#L109-L111

Reply:
ah right, makes it even simpler. will do.